core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -1208,3 +1226,36 @@ func findEthTxWithNonce(db *gorm.DB, fromAddress gethCommon.Address, nonce uint)
 	}
 	return &etx, errors.Wrap(err, "findEthTxsWithNonce failed")
 }
+
+func (ec *EthConfirmer) ResumePendingTaskRuns(ctx context.Context, head models.Head) error {
```

Comment:
I'd say this logic belongs in the caller, no?

Reply:
It's a bit awkward on either end since it's a join between `eth_tx*` tables and `pipeline_task_runs`. I ended up keeping it in EthConfirmer since the majority of the join logic is bptxm specific.